docs/ingestion/schema-design.md

```diff
@@ -261,7 +261,7 @@ native boolean types, Druid ingests these values as strings if `druid.expression
 the [array functions](../querying/sql-array-functions.md) or [UNNEST](../querying/sql-functions.md#unnest). Nested
 columns can be queried with the [JSON functions](../querying/sql-json-functions.md).
 
-We also highly recommend setting `druid.generic.useDefaultValueForNull=false` when using these columns since it also enables out of the box `ARRAY` type filtering. If not set to `false`, setting `sqlUseBoundsAndSelectors` to `false` on the [SQL query context](../querying/sql-query-context.md) can enable `ARRAY` filtering instead.
+We also highly recommend setting `druid.generic.useDefaultValueForNull=false` (the default) when using these columns since it also enables out of the box `ARRAY` type filtering. If not set to `false`, setting `sqlUseBoundsAndSelectors` to `false` on the [SQL query context](../querying/sql-query-context.md) can enable `ARRAY` filtering instead.
```

Comment:
Hmm. This is ripe for confusion, since it's the only place `sqlUseBoundsAndSelectors` is mentioned, and it doesn't say what the setting does. It also makes it sound like there is no real downside to setting it to `false` always. But there is a downside, right?

I am not sure mentioning the setting is worth the added confusion. How about we don't mention it at all, and have the official position in the docs be that you need SQL-compatible null handling in order to get `ARRAY` filtering?

Btw, I recognize this patch isn't really changing this section meaningfully. But… still.

Reply:
uh oh, this must have gotten lost in some document merging, since it was also documented on the query-context docs in #14760, but i no longer see it there...

Reply:
oh wait, i guess it is still there, but is a typo here, it should be `sqlUseBoundAndSelectors` not `sqlUseBoundsAndSelectors`

Reply:
anyway, yeah i can just remove mention of it i suppose, but it is documented at least

Reply:
thinking a bit more about this, since it is the default now I think i can just remove this entirely, and doing some testing it isn't even true. If this flag isn't set, then the filters plan into expression filters, which do produce the correct results, just a lot less efficiently than if the flag is set.